presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushAggregationIntoTableScan.java

```diff
@@ -119,7 +120,7 @@ public Result apply(AggregationNode node, Captures captures, Context context)
                 .map(Entry::getKey)
                 .collect(toImmutableList());
 
-        if (aggregateFunctions.isEmpty()) {
+        if (aggregateFunctions.isEmpty() && !aggregations.isEmpty()) {
```

Comment:
Apparently  `aggregateFunctions.size() == aggregations.size()`, so you don't need the `&&` part.

Reply:
I forgot that I no longer return empty list when an aggregate can not be transformed as we already check for only simple symbol references. This check makes no sense any more.